kratos/utilities/connectivity_preserve_modeler.h

```diff
@@ -129,15 +124,60 @@ class ConnectivityPreserveModeler : public Modeler
         }
         DestinationModelPart.AddConditions(temp_conditions.begin(), temp_conditions.end());
         
+        //ensure that the submodelparts are correctly filled
+        PopulateSubModelParts(OriginModelPart, DestinationModelPart);
+        
         //generating tables
 	DestinationModelPart.Tables() = OriginModelPart.Tables();
 
-
+        
+        
         Communicator::Pointer pComm = OriginModelPart.GetCommunicator().Create();
         DestinationModelPart.SetCommunicator(pComm);
 
         KRATOS_CATCH("");
     }
+    
+protected:
+    
+    void PopulateSubModelParts(
+        ModelPart& OriginModelPart,
+        ModelPart& DestinationModelPart)
+    {
+        KRATOS_WATCH(OriginModelPart)
+        KRATOS_WATCH(DestinationModelPart)
+        std::vector<ModelPart::NodeType::SizeType > ids;
+        for(auto part=OriginModelPart.SubModelPartsBegin(); part!=OriginModelPart.SubModelPartsEnd(); part++)
+        {
+            KRATOS_WATCH(part->Name())
+            
+            
+            auto& destination_part = DestinationModelPart.CreateSubModelPart(part->Name());
+            
+            //add nodes
+            ids.reserve(part->Nodes().size());
+            for(auto it=part->NodesBegin(); it!=part->NodesEnd(); ++it)
+                ids.push_back(it->Id());
+            destination_part.AddNodes(ids);
+
+            //add conditions
+            ids.clear();
+            ids.reserve(part->Conditions().size());
+            for(auto it=part->ElementsBegin(); it!=part->ElementsEnd(); ++it)
+                ids.push_back(it->Id());
+            destination_part.AddElements(ids);
+            
+            //add conditions
+            ids.clear();
+            ids.reserve(part->Conditions().size());
+            for(auto it=part->ConditionsBegin(); it!=part->ConditionsEnd(); ++it)
+                ids.push_back(it->Id());
+            destination_part.AddConditions(ids);   
```

Comment:
yes you are right about iterators (and watches were leftovers)

now corrected to use iterators